geode-core/src/distributedTest/java/org/apache/geode/cache/query/dunit/IndexUsageInJoinQueryDistributedTest.java

```diff
@@ -0,0 +1,209 @@
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.cache.query.dunit;
+
+import static org.apache.geode.cache.RegionShortcut.REPLICATE;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import org.apache.geode.cache.Cache;
+import org.apache.geode.cache.Region;
+import org.apache.geode.cache.client.ClientCache;
+import org.apache.geode.cache.client.ClientRegionShortcut;
+import org.apache.geode.cache.query.IndexExistsException;
+import org.apache.geode.cache.query.IndexNameConflictException;
+import org.apache.geode.cache.query.Query;
+import org.apache.geode.cache.query.QueryService;
+import org.apache.geode.cache.query.RegionNotFoundException;
+import org.apache.geode.distributed.internal.DistributionStats;
+import org.apache.geode.distributed.internal.InternalDistributedSystem;
+import org.apache.geode.pdx.PdxReader;
+import org.apache.geode.pdx.PdxSerializable;
+import org.apache.geode.pdx.PdxWriter;
+import org.apache.geode.test.dunit.rules.ClientVM;
+import org.apache.geode.test.dunit.rules.ClusterStartupRule;
+import org.apache.geode.test.dunit.rules.MemberVM;
+import org.apache.geode.test.junit.categories.OQLIndexTest;
+import org.apache.geode.test.junit.rules.serializable.SerializableTestName;
+
+@Category({OQLIndexTest.class})
+public class IndexUsageInJoinQueryDistributedTest implements Serializable {
+
+  private MemberVM locator;
+
+  private MemberVM server;
+
+  private ClientVM client;
+
+  private static final String PRODUCT_REGION_NAME = "product";
+
+  private static final String INSTRUMENT_REGION_NAME = "instrument";
+
+  @Rule
+  public ClusterStartupRule clusterRule = new ClusterStartupRule();
+
+  @Rule
+  public SerializableTestName testName = new SerializableTestName();
+
+  @Test
+  public void testSecondIndexUsedWhenExecutingJoinQuery() throws Exception {
+    // Start Locator
+    locator = clusterRule.startLocatorVM(0);
+
+    // Start server
+    server = clusterRule.startServerVM(1, locator.getPort());
+
+    // Create server regions
+    server.invoke(() -> createServerRegionsAndIndexes());
+
+    // Start client
+    client = clusterRule.startClientVM(2, c -> c.withLocatorConnection(locator.getPort()));
+
+    // Create client regions
+    client.invoke(() -> createClientRegions());
+
+    // Load regions
+    int numProducts = 1000, numInstruments = 100;
```

Comment:
There was some discussion on the dev channel to avoid single line multiple variable declarations. Maybe, can split it up 
```suggestion
    int numProducts = 1000;
    int numInstruments = 100;
```